File: hydrogen-v2/app/sql/models/hub.py
import datetime
import sqlalchemy
import sqlalchemy.orm as orm
import sqlalchemy.ext.declarative as decl

from werkzeug.security import generate_password_hash, check_password_hash
from flask_login import UserMixin

from app.sql.session import SqlAlchemyBase


class Hub(SqlAlchemyBase):

	INTVALUES = ["id", "owner_id", "game_id", "users_limit"]

	#def defAv():
	#	return open("./static/databases/images/avatars/def.png", "rb").read()

	__tablename__ = "hubs"

	id = sqlalchemy.Column(
		sqlalchemy.Integer,
		primary_key=True,
		autoincrement=True
	)

	name = sqlalchemy.Column(
		sqlalchemy.String,
		unique = True
	)

	description = sqlalchemy.Column(
		sqlalchemy.String(),
		default="No description"
	)

	owner_id = sqlalchemy.Column(
		sqlalchemy.Integer, sqlalchemy.ForeignKey("users.id")
	)
	owner = orm.relation("User", lazy='subquery')

	users_limit = sqlalchemy.Column(sqlalchemy.Integer)

	public = sqlalchemy.Column(sqlalchemy.Boolean, default=True)

	password_hash = sqlalchemy.Column(sqlalchemy.String)

	game_id = sqlalchemy.Column(
		sqlalchemy.Integer, sqlalchemy.ForeignKey("games.id")
	)
	game = orm.relation("Game", lazy='subquery')

	connections = orm.relation("Connection", back_populates="hub", lazy='subquery')

	def set_password(self, password):
		self.password_hash =generate_password_hash(password)

	def check_password(self, password):
		if self.public: return True
		return check_password_hash(self.password_hash, password)

	def __repr__(self): return f"<Hub> {self.id} {self.name}"

	def to_dict(self):
		return {
			"type": "<Hub>",
			"id": self.id,
			"name": self.name,
			"game": self.game.to_dict(),
			"owner": self.owner
		}

	def get_json(self):
		from app.sql import session as Session
		return {
			"id": self.id,
			"name": self.name,
			"owner": self.owner.login,
			"description": self.description,
			"game": self.game.get_json(),
			"is_public": self.public,
			"max_players": self.users_limit,
			"players": len(Session.create_session().query(Hub).filter(Hub.id==self.id).all()	)
		}

	def to_json(self): return self.get_json()

File: application/app/src/main/java/com/example/hydro/request/models/Hub.java
package com.example.hydro.request.models;

public class Hub {

    public int id; // : self.id
    public String name; // : self.name
    public String owner; // : self.owner.login
    public String description; // : self.description
    public Game game; // : self.game.get_json()
    public boolean is_public; //: self.public
    public int max_players; // : self.users_limit
    public int players; // : len(self.connections),

    public Hub(String name){
        this.name = name;
    }

    public Hub(int id,
               String name,
               String owner,
               String description,
               Game game,
               boolean is_public,
               int max_players,
               int players){

        this.id = id;
        this.name = name;
        this.owner = owner;
        this.description = description;
        this.game = game;
        this.is_public = is_public;
        this.max_players = max_players;
        this.players = players;
    }

    public String getName() {
        return this.name;
    }
}
